File: Codigos/Ejercicio10.java
import java.util.LinkedList;
import java.util.Scanner;
 
public class Ejercicio10 {
    public static void main(String[] args) {
        LinkedList<String> fila = new LinkedList<>();
        Scanner sc = new Scanner(System.in);
 
        System.out.print("¿Cuantos clientes en la fila inicial? ");
        int n = sc.nextInt();
        for (int i = 0; i < n; i++) {
            System.out.print("Cliente " + (i+1) + ": ");
            fila.add(sc.next());
        }
 
        System.out.println("\n--- SIMULACION DE ATENCION ---");
        while (!fila.isEmpty()) {
            System.out.println("Atendiendo a: " + fila.removeFirst());
            if (!fila.isEmpty())
                System.out.println("Siguiente: " + fila.getFirst());
            System.out.println("Clientes restantes: " + fila.size());
        }
        System.out.println("\nFila de atencion finalizada.");
        sc.close();
    }
}

File: Codigos/Ejercicio10.cpp
#include <iostream>
#include <list>
#include <string>
using namespace std;
 
int main() {
    list<string> fila;
    int n; string cliente;

    cout << "Fila de clientes usando LinkedList/list" << endl;
    cout << "¿Cuantos clientes en la fila inicial? "; cin >> n;
    for (int i = 0; i < n; i++) {
        cout << "Cliente " << i+1 << ": "; cin >> cliente;
        fila.push_back(cliente);
    }
 
    cout << "\n--- SIMULACION DE ATENCION ---" << endl;
    while (!fila.empty()) {
        cout << "Atendiendo a: " << fila.front() << endl;
        fila.pop_front();
        if (!fila.empty())
            cout << "Siguiente en fila: " << fila.front() << endl;
        cout << "Clientes restantes: " << fila.size() << endl;
    }
    cout << "\nFila de atencion finalizada." << endl;
    return 0;
}
